pallets/passkey/src/lib.rs

```diff
@@ -232,3 +265,67 @@ where
 		passkey_nonce.pre_dispatch(&who, &some_call.clone().into(), info, 0usize)
 	}
 }
+
+/// Passkey related tx payment
+#[derive(Encode, Decode, Clone, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct ChargeTransactionPayment<T: Config>(pub T::AccountId, pub Call<T>);
+
+impl<T: Config> ChargeTransactionPayment<T>
+where
+	<T as frame_system::Config>::RuntimeCall:
+		From<Call<T>> + Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
+{
+	/// Validates the transaction fee paid with tokens.
+	pub fn pre_dispatch(&self) -> Result<(), TransactionValidityError> {
+		let info = &self.1.get_dispatch_info();
+		let len = self.0.using_encoded(|c| c.len());
+		let runtime_call: <T as frame_system::Config>::RuntimeCall =
+			<T as frame_system::Config>::RuntimeCall::from(self.1.clone());
+		let who = self.0.clone();
+		self.withdraw_token_fee(&who, &runtime_call, info, len, Zero::zero())?;
```

Comment:
I have a question regarding the code. Could you please explain why we don’t call pre_dispatch here?

Reply:
`ChargeTransactionPayment` does not expose those functions `SignedExtension` trait does so we cant directly use it here

Hence we had to do this redundant implementation of check nonce and this

Reply:
I think SignedExtension is implemented on ChargeTransactionPayment. I think we might have a name collision?

Reply:
Will give it try

Reply:
At first glance, it is duable but breaks benchmarks, figuring that out as a separate work

https://github.com/frequency-chain/frequency/issues/2051

yay?  Nay?

Reply:
Yay.